lib/auth/signup-handler.ts

```
@@ -0,0 +1,67 @@
+ * @param {string} args.email
+ * @param {string} args.password
+ * @param {string} args.ip
+ * @param {object} args.supabaseClient - Must have .auth.signUp({ email, password, options })
+ * @param {object} args.analytics - Must have .capture({ event, properties })
+ * @returns {Promise<{ status: number, body: any }>}
+ */
+export async function signupBusinessLogic({ email, password, ip, supabaseClient, analytics }: {
+  email: string;
+  password: string;
+  ip: string;
+  supabaseClient: { auth: { signUp: Function } };
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)